verilog/parser/verilog.y

```diff
@@ -4598,12 +4598,14 @@ expr_primary_parens
   ;
 expr_primary_braces
   : '{' '}'
-    { $$ = MakeTaggedNode(N::kExpression, MakeBraceGroup($1, nullptr, $2)); }
+    { $$ = MakeTaggedNode(N::kExpression,
+                          MakeTaggedNode(N::kConcatenationExpression, $1, nullptr, $2)); }
   | '{' value_range '{' expression_list_proper '}' '}'
-    { $$ = MakeTaggedNode(N::kExpression, $1, $2, MakeBraceGroup($3, $4, $5), $6); }
+    { $$ = MakeTaggedNode(N::kExpression, $1, $2, MakeTaggedNode(N::kConcatenationExpression, $3, $4, $5), $6); }
     /* repeat concatenation: $2 should be an expression, not a range */
-  | range_list_in_braces
-    { $$ = MakeTaggedNode(N::kExpression, $1); }
+  | '{' open_range_list '}'
```

Comment:
(duplicate suggestion from conversation comment):
I would instead live the rule using `range_list_in_braces`, but then re-tag `$1`, replacing `kBraceGroup` with `kConcatenationExpression` (if removing `kBraceGroup` is what you intend).